contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,488 @@
+    }
+
+    /**
+        @notice Return arrays of donor's recipients and deposit amounts (gOHM value based on
+                sOHM equivalent deposit), matched by index
+        @param donor_ Address of user donating yield
+    */
+    function getAllDeposits(address donor_) external view returns (address[] memory, uint256[] memory) {
+        uint256[] storage depositIds = depositorIds[donor_];
+
+        uint256 len = depositIds.length == 0 ? 1 : depositIds.length;
+
+        address[] memory addresses = new address[](len);
+        uint256[] memory agnosticDeposits = new uint256[](len);
+
+        if (depositIds.length == 0) {
+            addresses[0] = address(0);
+            agnosticDeposits[0] = 0;
+        } else {
+            for (uint256 index = 0; index < len; ++index) {
+                addresses[index] = recipientLookup[depositIds[index]];
+                agnosticDeposits[index] = _toAgnostic(depositInfo[depositIds[index]].principalAmount);
+            }
+        }
+
+        return (addresses, agnosticDeposits);
+    }
+
+    /**
+        @notice Return total amount of gOHM donated to recipient since last full redemption
+        @param donor_ Address of user donating yield
+        @param recipient_ Address of user recieiving donated yield
+    */
+    function donatedTo(address donor_, address recipient_) external view returns (uint256) {
+        uint256[] storage depositIds = depositorIds[donor_];
+
+        for (uint256 index = 0; index < depositIds.length; ++index) {
+            if (recipientLookup[depositIds[index]] == recipient_) {
+                return redeemableBalance(depositIds[index]);
+            }
+        }
+
+        return 0;
+    }
+
+    /**
+        @notice Return total amount of gOHM donated from donor since last full redemption
+        @param donor_ Address of user donating yield
+    */
+    function totalDonated(address donor_) external view returns (uint256) {
+        uint256[] storage depositIds = depositorIds[donor_];
+
+        uint256 principalTotal = 0;
+        uint256 agnosticTotal = 0;
+
+        for (uint256 index = 0; index < depositIds.length; ++index) {
+            DepositInfo storage currDeposit = depositInfo[depositIds[index]];
+
+            principalTotal += currDeposit.principalAmount;
+            agnosticTotal += currDeposit.agnosticAmount;
+        }
+
+        return _getOutstandingYield(principalTotal, agnosticTotal);
+    }
+
+    function getDepositorIds(address donor_) external view returns (uint256[] memory) {
+        return depositorIds[donor_];
+    }
+
+    /************************
+     * Recipient Functions
+     ************************/
+
+    /**
+        @notice Get redeemable gOHM balance of a specific deposit
+        @param depositId_ Deposit ID for this donation
+    */
+    function redeemableBalance(uint256 depositId_) public view returns (uint256) {
+        DepositInfo memory currDeposit = depositInfo[depositId_];
```

Comment:
have u tested using memory here instead of storage makes it more gas efficient?
i remember mucking around with the old yieldirector with doing stuff like this and it actually made it use more gas. 

From my understanding instantiating a memory costs more gas than instantiating storage since storage is a pointer, memory is an actual copy. But memory becomes more efficient if we perform alot of reads on it. 

So i wouldn't be surprised if totalRedeemableBalance() is more gas efficient using memory but im not so sure about this function since it reads it twice.